src/silicone/database_crunchers/lead_gas.py

```diff
@@ -170,6 +169,8 @@ def get_values_in_key_timepoint(idf):
 
             output_ts["variable"] = variable_follower
             output_ts["unit"] = data_follower_unit
+            for col in in_iamdf.extra_cols:
+                output_ts[col] = ""
```

Comment:
using an empty string is slightly dangerous, do we have a better solution? What problem is this solving (i.e. can you point me to the test that would fail if we removed these lines)?

Reply:
I find that I am unable to join the output onto the input unless I ensure that there is such a column (pyam won't append dfs with different numbers of columns and gives a weird error message). I test appending in e.g. test_derive_relationship_averaging_info.

Reply:
Should it only do this filling where values are missing, rather than overwriting any existing values?

Reply:
If I understand your point, this is to make new columns, they don't exist prior to this line.

Reply:
Yes but the way this is written, any new columns are always empty. My question is whether they can be taken from the lead database. The test below is what I was thinking

```python
@pytest.mark.parametrize(
        "to_infill", (
            pd.DataFrame(
                [
                    ["ma", "sa", "World", "Emissions|HFC|C2F6", "kt C2F6/yr", "", "SCM", 1, 0.5],
                    ["ma", "sb", "World", "Emissions|HFC|C2F6", "kt C2F6/yr", "", "SCM", 3, 3],
                    ["mb", "sb", "World", "Emissions|HFC|C2F6", "kt C2F6/yr", "", "", 2, 4],
                ],
                columns=[
                    "model", "scenario", "region", "variable", "unit", "meta", "climate_model", 2010, 2015]
                ,
            ),
        ),
    )
    def test_derive_relationship_extra_cols(self, test_db, to_infill):
        # test that extra columns in the lead dataframe are maintained while infilling
        tcruncher = self.tclass(test_db)
        variable_follower = "Emissions|HFC|C5F12"
        variable_leader = ["Emissions|HFC|C2F6"]
        cruncher = tcruncher.derive_relationship(
            variable_follower, variable_leader
        )

        to_infill = self._adjust_time_style_to_match(IamDataFrame(to_infill), test_db)
        infilled = cruncher(to_infill)

        def _get_comp_df(in_iamdf):
            tmp = in_iamdf.timeseries()
            comp_idx_to_drop = list(set(tmp.index.names) - {"model", "scenario"})
            tmp.index = tmp.index.droplevel(comp_idx_to_drop)

            return tmp

        np.testing.assert_allclose(
            _get_comp_df(infilled),
            _get_comp_df(to_infill).multiply(3.14 / _get_comp_df(to_infill)[2015], axis=0),
        )

        # Test that the result can be appended without problems.
        to_infill.append(infilled)

        # test that indexes are copied over as desired
        pd.testing.assert_index_equal(
            to_infill.timeseries().index.droplevel(["variable", "unit", "region"]),
            infilled.timeseries().index.droplevel(["variable", "unit", "region"]),
        )
```

It currently fails because the `infilled` SCM values are not copied from `to_infill` i.e.

```python
(Pdb) to_infill.timeseries()  # has 'SCM' in climate_model column
                                                                        2010  2015
model scenario region variable           unit       meta climate_model            
ma    sa       World  Emissions|HFC|C2F6 kt C2F6/yr      SCM             1.0   0.5
      sb       World  Emissions|HFC|C2F6 kt C2F6/yr      SCM             3.0   3.0
mb    sb       World  Emissions|HFC|C2F6 kt C2F6/yr                      2.0   4.0
(Pdb) infilled.timeseries()  # all 'climate_model' column is empty
                                                                          2010  2015
model scenario region variable            unit        meta climate_model            
ma    sa       World  Emissions|HFC|C5F12 kt C5F12/yr                     6.28  3.14
      sb       World  Emissions|HFC|C5F12 kt C5F12/yr                     3.14  3.14
mb    sb       World  Emissions|HFC|C5F12 kt C5F12/yr                     1.57  3.14
```

I'm not sure what the desired behaviour is, so just wanted to check if this looks right to you.

Reply:
I keep asking what people want to happen with meta and not really getting a clear answer - there's been the idea floating around that it should be set to 'infilled' or similar, but no consensus and no obvious reason that it should be a clone of the input, since it doesn't exactly come from any of the same sources.

Reply:
I think the answer is it's a mess haha. In that case let's leave it as is (i.e. just fill with spaces) and chuck a note in a docstring wherever you think is best. 

Last question, do we need to check this behaviour for other infillers or does it only apply to this one?

Reply:
We should probably add tests for other infillers, this was the only one I noticed it not working for. I'll add it to my fairly long to-do list.

Reply:
ye chuck up an issue and let's move on